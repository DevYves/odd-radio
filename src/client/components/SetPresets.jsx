import React, { Component } from 'react';

class SetPresets extends Component {

  render() {
    return (
      <div className="container presets">
        <div className="center presets-title">
          { this.props.title }
        </div>
        <div className="row border">
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.details, 'one')}>
            <div>1:</div>
            <p>{this.props.presets.one.name ? this.props.presets.one.name : 'Empty'}</p>
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.details, 'two')}>
            <div>2:</div>
            <p>{this.props.presets.two.name ? this.props.presets.two.name : 'Empty'}</p>
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.details, 'three')}>
            <div>3:</div>
            <p>{this.props.presets.three.name ? this.props.presets.three.name : 'Empty'}</p>
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.details, 'four')}>
            <div>4:</div>
            <p>{this.props.presets.four.name ? this.props.presets.four.name : 'Empty'}</p>
          </div>
          <div className="five columns presets-station" onClick={(e) => this.props.clickFunction(this.props.details, 'five')}>
            <div>5:</div>
            <p>{this.props.presets.five.name ? this.props.presets.five.name : 'Empty'}</p>
          </div>
        </div>
      </div>
    );
  }
}

export default SetPresets;