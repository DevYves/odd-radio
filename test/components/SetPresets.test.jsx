import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import SetPresets from '../../src/client/components/SetPresets';

describe('Set Presets Component', () => {
    const props = {
    presets: {
      one: {name: 'CMYK'},
      two: {name: 'CMYK'},
      three: {name: 'CMYK'},
      four: {name: 'CMYK'},
      five: {name: 'CMYK'}
    },
    title:"Save Station in Presets",
    clickFunction: jest.fn(),
    details: {
       id: '1',
       name: 'QWER',
       audioFeed: 'www.google.ca',
       streamType: 'audio/mp3',
       frequency: '91.2 FM',
       city: 'Hamilton',
       province: 'ON'
     }
    }

    const props2 = {
      presets: {
      one: {},
      two: {},
      three: {},
      four: {},
      five: {}
    },
    title:"Save Station in Presets",
    clickFunction: jest.fn(),
    details: {
       id: '1',
       name: 'QWER',
       audioFeed: 'www.google.ca',
       streamType: 'audio/mp3',
       frequency: '91.2 FM',
       city: 'Hamilton',
       province: 'ON'
     }

    }

  const subject = <SetPresets { ...props } />;
  const subject2 = <SetPresets { ...props2} />

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('when preset one is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.presets-station').at(0).simulate('click');

    expect(props.clickFunction).toHaveBeenCalled();
  })

  describe('when preset two is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.presets-station').at(1).simulate('click');
    expect(props.clickFunction).toHaveBeenCalled();

  })

  describe('when preset three is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.presets-station').at(2).simulate('click');
    expect(props.clickFunction).toHaveBeenCalled();

  })

  describe('when preset four is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.presets-station').at(3).simulate('click');
    expect(props.clickFunction).toHaveBeenCalled();

  })

  describe('when preset five is clicked', () => {
    const wrapper = shallow(subject);
    wrapper.find('.presets-station').at(4).simulate('click');
    expect(props.clickFunction).toHaveBeenCalled();

  })

  describe('on load has five <p> tags', () => {
    const wrapper = shallow(subject);
    expect(wrapper.find('p').length).toBe(5);
  })

  describe('when station 1 is saved to the preset should display that stations name', () => {
    const wrapper = shallow(subject);
    console.log(wrapper.find('p').at(0).text());
    expect(wrapper.find('p').at(0).text()).toBe('CMYK');
  })

  describe('when a station 1 is not saved to the preset should display empty', () => {
    const wrapper = shallow(subject2);
    expect(wrapper.find('p').at(0).text()).toBe('Empty');
  })

  describe('when a station 2 is saved to the preset should display that stations name', () => {
    const wrapper = shallow(subject);
    console.log(wrapper.find('p').at(1).text());
    expect(wrapper.find('p').at(1).text()).toBe('CMYK');
  })

    describe('when a station 2 is not saved to the preset should display empty', () => {
    const wrapper = shallow(subject2);
    expect(wrapper.find('p').at(1).text()).toBe('Empty');
  })






});