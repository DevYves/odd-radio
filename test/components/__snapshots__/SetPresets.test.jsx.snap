// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set Presets Component matches snapshot 1`] = `
<div
  className="container presets"
>
  <div
    className="center row presets-title"
  >
    Save Station in Presets
  </div>
  <div
    className="row border"
  >
    <div
      className="five columns presets-station"
      onClick={[Function]}
    >
      <div>
        1:
      </div>
      <p>
        CMYK
      </p>
    </div>
    <div
      className="five columns presets-station"
      onClick={[Function]}
    >
      <div>
        2:
      </div>
      <p>
        CMYK
      </p>
    </div>
    <div
      className="five columns presets-station"
      onClick={[Function]}
    >
      <div>
        3:
      </div>
      <p>
        CMYK
      </p>
    </div>
    <div
      className="five columns presets-station"
      onClick={[Function]}
    >
      <div>
        4:
      </div>
      <p>
        CMYK
      </p>
    </div>
    <div
      className="five columns presets-station"
      onClick={[Function]}
    >
      <div>
        5:
      </div>
      <p>
        CMYK
      </p>
    </div>
  </div>
</div>
`;
